core/src/test/java/org/elasticsearch/VersionTests.java

```diff
@@ -230,7 +232,7 @@ public void testParseVersion() {
         });
         assertSame(Version.CURRENT, Version.fromString(Version.CURRENT.toString()));
 
-        assertSame(Version.fromString("2.0.0-SNAPSHOT"), Version.fromString("2.0.0"));
+        assertEquals(Version.fromString("2.0.0-SNAPSHOT"), Version.fromId(2000099));
```

Comment:
Both options work for me, just wanted to make sure you did not do that because of a bug in fromString.